python/cudf/cudf/core/scalar.py

```diff
@@ -16,7 +17,56 @@
 )
 
 
-class Scalar(BinaryOperand):
+# Note that the metaclass below can easily be generalized for use with
+# other classes, if needed in the future. Simply replace the arguments
+# of the `__call__` method with `*args` and `**kwargs`. This will
+# result in additional overhead when constructing the cache key, as
+# unpacking *args and **kwargs is not cheap. See the discussion in
+# https://github.com/rapidsai/cudf/pull/11246#discussion_r955843532
+# for details.
+class CachedScalarInstanceMeta(type):
+    """
+    Metaclass for Scalar that caches `maxsize` instances.
+
+    After `maxsize` is reached, evicts the least recently used
+    instances to make room for new values.
+    """
+
+    def __new__(cls, names, bases, attrs, **kwargs):
+        return type.__new__(cls, names, bases, attrs)
+
```

Comment:
Can't this be deleted? Or will it fail because by default `__new__` can't handle the arbitrary kwargs?

Reply:
Nope - we need to override `__new__` to include the extra argument.